src/convenience/webhook.ts

```diff
@@ -252,27 +250,30 @@ function webhookCallback<
 
 function timeoutIfNecessary(
     task: Promise<void>,
-    onTimeout: "throw" | "ignore" | (() => unknown),
     timeout: number,
+    onTimeout?: () => unknown, // need async?
 ): Promise<void> {
     if (timeout === Infinity) return task;
-    return new Promise((resolve, reject) => {
-        const handle = setTimeout(() => {
-            debugErr(`Request timed out after ${timeout} ms`);
-            if (onTimeout === "throw") {
-                reject(new Error(`Request timed out after ${timeout} ms`));
-            } else {
-                if (typeof onTimeout === "function") onTimeout();
-                resolve();
-            }
-            const now = Date.now();
-            task.finally(() => {
-                const diff = Date.now() - now;
-                debugErr(`Request completed ${diff} ms after timeout!`);
-            });
-        }, timeout);
-        task.then(resolve)
-            .catch(reject)
-            .finally(() => clearTimeout(handle));
-    });
+    const { promise, resolve, reject } = Promise.withResolvers<void>();
+
+    const handle = setTimeout(() => {
+        debugErr(`Request timed out after ${timeout} ms`);
+        if (onTimeout) {
```

Comment:
```suggestion
        if (onTimeout !== undefined) {
```
just to be more explicit